tracer/src/Datadog.Trace/Ci/Agent/CIWriterHttpSender.cs

```diff
@@ -171,15 +172,35 @@ private async Task SendPayloadAsync<T>(EventPlatformPayload payload, Func<IApiRe
 
         private async Task SendPayloadAsync(CIVisibilityProtocolPayload payload)
         {
-            var payloadArray = payload.ToArray();
-            if (Log.IsEnabled(LogEventLevel.Debug))
+            byte[] payloadArray;
+            if (!payload.UseEvpProxy)
             {
-                Log.Debug<int, string>("Sending ({NumberOfTraces} events) {BytesValue} bytes...", payload.Count, payloadArray.Length.ToString("N0"));
+                // If we are in agentless mode (no EVP Proxy) then we use gzip compression, supported by the intake
+                var mStream = new MemoryStream();
```

Comment:
Technically not required, I know, but feels wrong not disposing these things, no? 😅 
```suggestion
                using var mStream = new MemoryStream();
```

Reply:
done in [5ae76b2](https://github.com/DataDog/dd-trace-dotnet/pull/4637/commits/5ae76b2768a46bbb0b71a486ab30e1614305138b)